testing_and_setup/compass/ocean/jigsaw_to_MPAS/build_mesh.py

```diff
@@ -31,32 +31,49 @@
 
 
 def build_mesh(preserve_floodplain=False, floodplain_elevation=20.0,
-               do_inject_bathymetry=False):
+               do_inject_bathymetry=False, geometry='sphere'):
 
-    print('Step 1. Build cellWidth array as function of latitude and '
-          'longitude')
-    cellWidth, lon, lat = define_base_mesh.cellWidthVsLatLon()
-    da = xarray.DataArray(cellWidth,
+    if geometry=='sphere':
+        on_sphere=True
+    else:
+        on_sphere=False
+
+    print('Step 1. Build cellWidth array as function of horizontal coordinates')
+    if on_sphere:
+       cellWidth, lon, lat = define_base_mesh.cellWidthVsLatLon()
+       da = xarray.DataArray(cellWidth,
                           dims=['lat','lon'],
                           coords={'lat':lat,'lon':lon},
                           name='cellWidth')
-    da.to_netcdf('cellWidthVsLatLon.nc')
-    
+       cw_filename='cellWidthVsLatLon.nc'
+       da.to_netcdf(cw_filename)
+    else:
+       cellWidth, x, y, geom_points, geom_edges = define_base_mesh.cellWidthVsXY()
+       da = xarray.DataArray(cellWidth,
+                          dims=['y','x'],
+                          coords={'y':y,'x':x},
+                          name='cellWidth')
+       cw_filename='cellWidthVsXY.nc'
+       da.to_netcdf(cw_filename)
```

Comment:
`da.to_netcdf(cw_filename)` could probably be moved outside the if structure here so it's only in one place, but it's also totally fine the way it is.

Reply:
I considered that but thought it might look slightly less readable if that line was dangling after the if-construct.  I figured this way each branch is fully self-contained.

Reply:
Perfect, I agree. I hesitated to even bring it up.

Reply:
It's not a great idea to create a variable (`da` in this case) within an if/else and then use it outside unless it's really necessary so I agree that it's better to leave as it is.